lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h

```diff
@@ -0,0 +1,1001 @@
+  std::string pei;
+  uint64_t udmGroupId;   // NfGroupId
+  uint64_t ausfGroupId;  // NfGroupId;
+  std::string routingIndicator;
+  std::string drxParameter;
+  std::string subRfsp;
+  uint32_t usedRfsp;  // RfspIndex type;
+  ambr_t subUeAmbr;
+  bool smsSupport;
+  std::string smsfId;  // NfInstanceId type
+  std::string
+      seafData;  // SeafData which will come while AUSF communication for AUTN.
+  std::string pcfId;                   // NfInstanceId
+  std::string pcfAmPolicyUri;          // Uri
+  std::string pcfUePolicyUri;          // Uri
+  std::string hpcfId;                  // NfInstanceId
+  std::string restrictedRatList;       // array(RatType)
+  std::string forbiddenAreaList;       // array(Area)
+  std::string serviceAreaRestriction;  // ServiceAreaRestriction
+  std::string restrictedCnList;        // array(CoreNetworkType)
+  std::string eventSubscriptionList;   // array(AmfEventSubscription)
+  std::string mmContextList;           // array(MmContext)
+  std::string sessionContextList;      // array(PduSessionContext)
+  std::string traceData;               // TraceData
+
+  /* SCTP assoc id */
+  sctp_assoc_id_t sctp_assoc_id_key;
+
+  /* gNB UE NGAP ID,  Unique identity the UE within gNodeB */
+  gnb_ue_ngap_id_t gnb_ue_ngap_id : 24;
+  gnb_ngap_id_key_t gnb_ngap_id_key;
+
+  /* MME UE S1AP ID, Unique identity the UE within MME */
+  amf_ue_ngap_id_t amf_ue_ngap_id;
+};
+
+class amf_app_ue_context : public amf_ue_context_t, public ue_m5gmm_context_s {
+ public:
+  amf_app_ue_context() {}
+  ~amf_app_ue_context() {}
+  // check & create state information.
+  int amf_insert_ue_context(
+      const amf_ue_context_t* amf_ue_context,
+      const ue_m5gmm_context_s* ue_context_p);
+  amf_ue_ngap_id_t amf_app_ctx_get_new_ue_id(
+      amf_ue_ngap_id_t* amf_app_ue_ngap_id_generator_p);
+  // Notify NGAP about the mapping between amf_ue_ngap_id and
+  // sctp assoc id + gnb_ue_ngap_id
+  void notify_ngap_new_ue_amf_ngap_id_association(
+      const ue_m5gmm_context_s* ue_context_p);
+  void amf_remove_ue_context(
+      amf_ue_context_t* const amf_ue_context,
+      ue_m5gmm_context_s* const ue_context_p);
+};
+
+ue_m5gmm_context_s* amf_create_new_ue_context(void);
+
+/* Retrive required UE context from respective hash table*/
+amf_context_t* amf_context_get(const amf_ue_ngap_id_t ue_id);
+ue_m5gmm_context_s* amf_ue_context_exists_amf_ue_ngap_id(
+    const amf_ue_ngap_id_t amf_ue_ngap_id);
+int amf_context_upsert_imsi(amf_context_t* elm) __attribute__((nonnull));
+
+void amf_ctx_set_valid_imsi(
+    amf_context_t* ctxt, imsi_t* imsi, const imsi64_t imsi64)
+    __attribute__((nonnull)) __attribute__((flatten));
+
+void amf_ctx_set_attribute_valid(
+    amf_context_t* ctxt, const uint32_t attribute_bit_pos)
+    __attribute__((nonnull)) __attribute__((flatten));
+
+void amf_ctx_set_attribute_present(
+    amf_context_t* ctxt, const int attribute_bit_pos) __attribute__((nonnull))
+__attribute__((flatten));
```

Comment:
provide comments about the function usage

Reply:
fixed